### ISSUE TITLE
Grid2D doesn't work well with integer dx

### ISSUE THREAD

**Issue by guyer** [usnistgov/fipy#672]
```python
import fipy as fp

nx = 20
ny = nx
dx = 1
dy = dx
mesh = fp.Grid2D(dx=dx, dy=dy, nx=nx, ny=ny)

phi = fp.CellVariable(mesh=mesh, value=0.)
eq = fp.TransientTerm() == fp.DiffusionTerm(coeff=1)

eq.solve(var=phi, dt=1.)
```
raises 
* `RuntimeError: Factor is exactly singular`
* `StagnatedSolverWarning: The solver stagnated. Iterations: 2. Relative error: nan`
* `RuntimeWarning: invalid value encountered in true_divide`

depending on solver. 

`dx = 1.` fixes this.

**Comment by raviapatel** [comment]
In general i find this also problem when initializing cell variables with an integer it raises warning and solutions are not right. just doing "1." fixes this problem. This is issue even with python3!

**Comment by guyer** [comment]
@raviapatel: That's true and unlikely to change. There are internal reasons why cell variables need to be able to hold integers, but integers are not suitable for solving PDEs.
